internal/provider/vaultradar/resource_radar_source_github_enterprise.go

```diff
@@ -0,0 +1,235 @@
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.RegexMatches(regexp.MustCompile(`^[a-zA-Z0-9-_.]+$`),
+						"must contain only letters, numbers, hyphens, underscores, or periods",
+					),
+				},
+			},
+			"token": schema.StringAttribute{
+				Description: "GitHub personal access token.",
+				Required:    true,
+				Sensitive:   true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			// Optional inputs
+			"project_id": schema.StringAttribute{
+				Description: "The ID of the HCP project where Vault Radar is located. If not specified, the project specified in the HCP Provider config block will be used, if configured.",
+				Optional:    true,
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+		},
+	}
+}
+
+type gitHubEnterpriseSource struct {
+	ID                 types.String `tfsdk:"id"`
+	DomainName         types.String `tfsdk:"domain_name"`
+	GitHubOrganization types.String `tfsdk:"github_organization"`
+	Token              types.String `tfsdk:"token"`
+	ProjectID          types.String `tfsdk:"project_id"`
+}
+
+func (r *sourceGitHubEnterpriseResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+	client, ok := req.ProviderData.(*clients.Client)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *clients.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+	r.client = client
+}
+
+func (r *sourceGitHubEnterpriseResource) ModifyPlan(ctx context.Context, req resource.ModifyPlanRequest, resp *resource.ModifyPlanResponse) {
+	modifiers.ModifyPlanForDefaultProjectChange(ctx, r.client.Config.ProjectID, req.State, req.Config, req.Plan, resp)
+}
+
+func (r *sourceGitHubEnterpriseResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan gitHubEnterpriseSource
+
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	projectID := r.client.Config.ProjectID
+	if !plan.ProjectID.IsUnknown() {
+		projectID = plan.ProjectID.ValueString()
+	}
+
+	res, err := clients.OnboardRadarSource(ctx, r.client, projectID, radar_service.OnboardDataSourceBody{
+		Type:          "github_enterprise",
+		Name:          plan.GitHubOrganization.ValueString(),
+		ConnectionURL: plan.DomainName.ValueString(),
+		Token:         plan.Token.ValueString(),
+	})
+	if err != nil {
+		resp.Diagnostics.AddError("Error creating Radar source", err.Error())
+		return
+	}
+
+	plan.ID = types.StringValue(res.GetPayload().ID)
```

Comment:
@jasonpilz Note that I'm using the GUID generated from the server call, instead of generating something and setting the id first, which seem to be inferred by the new resources guide.

Is this okay?